### PR TITLE
Create user and feed microservices

### PR DESCRIPTION
**Pull request by kovaj024**

### CODE REVIEW

File: udagram-api-feed/docker-compose.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-      POSTGRES_USER: postgres
-      POSTGRES_PASSWORD: postgres
-    volumes:
-      - postgres-data:/var/lib/postgresql/data
-
-volumes:
+version: '3.8'
+services:
+  postgres:
+    image: postgres:latest
```
Comment (victorgrubio) [inline udagram-api-feed/docker-compose.yml:4]: use a fixed tag version, avoid using latest for constant image update. You could leverage alpine as well to reduce image size: https://hub.docker.com/_/postgres/tags?page=1&name=alpine

File: udagram-api-feed/docker-compose.yml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    container_name: udagram-db
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_DB: udagram-db
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
```
Comment (victorgrubio) [inline udagram-api-feed/docker-compose.yml:11]: You could leverage env variables for the password, use and username. You can even substitute using ${ENV_VAR} if it's different. You could also use .env_files - https://docs.docker.com/compose/environment-variables/env-file/

File: udagram-api-user/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM node:13
```
Comment (victorgrubio) [inline udagram-api-user/Dockerfile:1]: You should use a more defined version, and leverage alpine https://hub.docker.com/_/node/tags?page=1&name=alpine

File: udagram-api-user/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+version: '3.8'
+services:
+  postgres:
+    image: postgres:latest
```
Comment (victorgrubio) [inline udagram-api-user/docker-compose.yml:4]: Again, the version should be better. Moreover, if you have already a docker-compose file in the root, there's no need to have duplicate individuals. You can just spin one service using `docker-compose up SERVICE_NAME -d`. Therefore, you can remove this file

File: udagram-api-user/src/controllers/v0/users/routes/auth.router.ts [hunk 5]
```diff
@@ -8,7 +8,6 @@ import * as jwt from 'jsonwebtoken';
 import {NextFunction} from 'connect';
 
 import * as EmailValidator from 'email-validator';
-import {config} from 'bluebird';
```
Comment (victorgrubio) [inline udagram-api-user/src/controllers/v0/users/routes/auth.router.ts:11]: Any reason for this? Was it imported by mistake?

File: docker-compose-build.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+version: "3"
```
Comment (victorgrubio) [inline docker-compose-build.yaml:1]: I'd rename this file to be `docker-compose.yaml`

File: udagram-frontend/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FROM beevelop/ionic:latest AS ionic
```
Comment (victorgrubio) [inline udagram-frontend/Dockerfile:1]: Again, avoid using latest

File: udagram-frontend/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+RUN npm ci
+
+COPY . .
+RUN ionic build
+
+FROM nginx:alpine
```
Comment (victorgrubio) [inline udagram-frontend/Dockerfile:11]: Use a fixed version

File: udagram-reverseproxy/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,2 @@
+FROM nginx:alpine
```
Comment (victorgrubio) [inline udagram-reverseproxy/Dockerfile:1]: Use a fixed nginx version

**Review by victorgrubio — CHANGES_REQUESTED** [review]
I'm going to try and run it right now. All changes until now are code quality oriented. Good job on the overall migration :)

**Comment by victorgrubio** [comment]
missing dockerignore in root folder. I suggest you use something like https://github.com/BretFisher/node-docker-good-defaults/blob/main/.dockerignore

File: docker-compose-build.yaml (hunk 6 above)
Comment (victorgrubio) [inline docker-compose-build.yaml:1]: Main docker-compose file is missing

File: udagram-api-user/src/controllers/v0/users/routes/auth.router.ts (hunk 5 above)
Comment (kovaj024) [reply udagram-api-user/src/controllers/v0/users/routes/auth.router.ts:11]: Unused import

**Review by victorgrubio — APPROVED** [review]
LGTM Good Job!

**Pull request merged**